include/pipettes/core/i2c_writer.hpp

```diff
@@ -38,32 +41,108 @@ class I2CWriter {
         queue->try_write(write_msg);
     }
 
-    void read(uint16_t device_address,
-              const Callback
-                  callback,  // NOLINT (performance-unnecessary-value-param)
-              uint8_t reg = 0x0) {
+    /**
+     * A single read to the i2c bus.
+     *
+     * @param device_address: device address on the i2c bus
+     * @param client_callback: the function that will write a CAN return message
+     * to the client queue.
+     * @param handle_callback: the function that will handle any data
+     * manipulation required for the specific task using the i2c bus. Takes in
+     * two buffers.
+     * @param reg: the register to read from, if relevant.
+     *
+     * A poll function automatically initiates a read/write to the specified
+     * register on the i2c bus.
+     */
+    void read(uint16_t device_address, SingleBufferTypeDef handle_callback,
+              SendToCanFunctionTypeDef client_callback, uint8_t reg = 0x0) {
         // We want to copy the callback every time as opposed to passing a
```

Comment:
can we remove this comment? it's not quite accurate (we're moving, not copying) and it also doesnt mention why we'd want to do this